include/pybind11/cast.h

```diff
@@ -973,7 +979,7 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
         return caster_t::cast(&src.get(), policy, parent);
     }
     template <typename T> using cast_op_type = std::reference_wrapper<type>;
-    operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type&(); }
+    operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type(); }
```

Comment:
```suggestion
    operator std::reference_wrapper<type>() { return cast_op<reference_t>(subcaster); }
```

or even (depending on the other discussion)

```suggestion
    operator std::reference_wrapper<type>() { return cast_op<type&>(subcaster); }
```

Reply:
So you already listed point 3/ where I removed a & based on review feedback as a "separate change"; how is this not the same?

Reply:
What do you mean? I don't mean this should be taken out, but if you go and change `std::add_lvalue_reference<type>` to `type &`, this part can also be simplified?

Reply:
If you don't agree with the suggestion in https://github.com/pybind/pybind11/pull/2705#discussion_r542041783, that's also fine with me, then just argue why not and do not do it!

Reply:
I mean `operator std::reference_wrapper<type>() { return cast_op<type &>(subcaster); }` is clearly a more substantive change than removing `&` from `operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type(); }`

Reply:
True, yes. I just saw this being used in other casters.

Reply:
I'm also fine with not applying this change; it's supposed to do exactly the same, though.

Reply:
Done.